elide-core/src/main/java/com/yahoo/elide/core/exceptions/CustomErrorException.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.exceptions;
+
+
+import com.yahoo.elide.core.ErrorObjects;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.Objects;
+
+/**
+ * Define your business exception extend this.
+ */
+public class CustomErrorException extends HttpStatusException {
```

Comment:
As you mention in #650 both the JSON API and GraphQL specs call for errors to be presented in this object format. Perhaps we should fold this into the `HttpStatusException` class so that any error thrown by Elide conforms to the spec.